### PR TITLE
feat(profile-routing): Add accountAddress param to user profile routing

### PR DESCRIPTION
**Pull request by ChefHutch**
- Modify profile url structure to receive `accountAddress` param
- Add support for rendering unconnected profiles

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -157,6 +161,9 @@ const App: React.FC = () => {
             <Route path="/collectibles">
               <Redirect to="/nfts" />
             </Route>
+            <Route path="/profile">
+              <Redirect to={`${nftsBaseUrl}/profile/${account?.toLowerCase() || nullProfileAccountAddress}`} />
```
Comment (ChefHutch) [inline src/App.tsx:165]: Set up redirect to users' profile (if connected), and a null address if not

File: src/state/nftMarket/helpers.ts [hunk 2]
```diff
@@ -498,3 +498,52 @@ export const combineNftMarketAndMetadata = (
...
 }
+
+/**
+ * Get in-wallet, on-sale & profile pic NFT metadata, complete with market data for a given account
+ * @param account
+ * @param collections
+ * @param profileNftWithCollectionAddress
+ * @returns Promise<NftToken[]>
+ */
+export const getCompleteAccountNftData = async (
```
Comment (ChefHutch) [inline src/state/nftMarket/helpers.ts:509]: This used to be in the reducer. Moved out as it's now used in two places.

File: src/views/Nft/market/Profile/components/ActivityHistory/ActivityPrice.tsx [hunk 3]
```diff
@@ -11,13 +11,18 @@ const ActivityPrice = ({ bnbBusdPrice, price }) => {
...
+                maximumFractionDigits: 5,
+              })}
+            </Text>
           </Flex>
           <Text fontSize="12px" color="textSubtle">
             {`(~$${priceInUsd.toLocaleString(undefined, {
               minimumFractionDigits: 2,
               maximumFractionDigits: 2,
-            })}`}
+            })})`}
```
Comment (ChefHutch) [inline src/views/Nft/market/Profile/components/ActivityHistory/ActivityPrice.tsx:25]: Bundling in tiny tweaks to the price. Not related to user routing 😬

File: src/views/Nft/market/Profile/index.tsx [hunk 4]
```diff
@@ -1,66 +1,16 @@
...
-          <ActivityHistory />
-        </Route>
-        <Route exact path={`${nftsBaseUrl}/profile`}>
-          <SubMenu />
-          <UserNfts />
-        </Route>
-      </Page>
-    </>
-  )
+  return <>{isConnectedProfile ? <ConnectedProfile /> : <UnconnectedProfile />}</>
```
Comment (ChefHutch) [inline src/views/Nft/market/Profile/index.tsx:13]: Profile root now returns either a connected profile or unconnected profile component. Core different between the two is that one makes use of profile/user nfts redux store, the other fetches on the fly.

File: src/App.tsx (hunk 1 above)
Comment (hachiojidev) [reply src/App.tsx:165]: This is not necessary. Simply make the address optional and handle at the view level.
```
<Route path="/profile/:address?" />
```

**Comment by ChefHutch** [comment]
:bonk: https://github.com/pancakeswap/pancake-frontend/pull/2397

**Pull request closed without merging**